### ISSUE TITLE
eCarePlanner reads patient data from SDS partitions (multiple provider EHRs)

### ISSUE THREAD

**Issue by kbertodatti** [chronic-care/mcc-project#364]
Decide: What to do about duplicate launch EHR/partition data.
Need Matt to provide SDS API to determine what partitions are there in order to read from them.

Matt's comments:
good question.  I don't know that HAPI FHIR provides an API to get partition names, so you might not be able to query the SDS for them per se.  that said, I think you could query for the Linkage resource from the patient's local partition, and that will contain references to patient records in other partitions.  I suspect you could use information from there to identify the partition names, but it might require a little parsing.  this is something we should all discuss to make sure we're on the same page about it

this is a test Linkage in our SDS database:
{"resourceType":"Linkage","item":[{"type":"source","resource":{"reference":"Patient/79ec6e57-3baa-44ab-a0a8-2fc78c5d58dc"}},{"type":"alternate","resource":{"reference":
"https://epicmobile.ohsu.edu/FHIRDEV/api/FHIR/R4/Patient/eFTHaVbQzCEwOEE97maN2MC2jJi-r8nnkhRh.umMUlz03"}}]}
so when you query that Linkage resource from the local partition, you can iterate over the "alternate" resource references, and parse out the partitions from them, which is always going to be the FHIR endpoint for the associated system.  in the case above, that would be https://epicmobile.ohsu.edu/FHIRDEV/api/FHIR/R4/

so basically, cut the reference string from 0 to just before the Patient/ bit, collect those strings into an array, and then reissue queries to the SDS for resources, passing in values from that array as X-Partition-Name headers
